projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -129,6 +129,23 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      */
     public totalItemCount: number = null;
 
+    /**
```

Comment:
While using functions is fine I like how ngFor defines the context better - https://github.com/angular/angular/blob/master/packages/common/src/directives/ng_for_of.ts#L14
It's defined in a separate class with getters for first/last/even/odd based on the index/count. This will allow for less code repetition and will be easier to maintain.

Reply:
Done